src/main.rs

```diff
@@ -33,6 +35,24 @@ const HEADING_RPC: &str = "OPTIONS (RPC)";
 struct Root {
     #[clap(subcommand)]
     cmd: Cmd,
+
```

Comment:
I would avoid adding arguments here unless needed globally.

Instead I would recommend making a new args struct that containing these and then using something like [config[(https://github.com/mehcode/config-rs). Then in any command that needs it you add 

```
#[flatten]
profile::Args
```

This avoids having to pass the extra args along e.g.
```
pub fn run(profiles_file: &std::path::PathBuf) -> Result<(), Error> {
```
becomes
```
pub fn run(&self) -> Result<(), Error> {
  self.profiles_file
```